### DISCUSSION TITLE
RFC should Flexslider be sunsetted for an alternative library?

### DISCUSSION THREAD

**Discussion by muskie9-opensource** [dynamic/silverstripe-flexslider#213] [Announcements]
**Overview**
After some digging, it seems the flexslider js library hasn't been actively supported since [Mar 7, '19](https://github.com/woocommerce/FlexSlider/commit/690832b7f972298e76e2965714657a2beec9e35c). The question arises as to whether this library is appropriate for long term support as a module on our end.

**Proposal**
Explore alternatives to Flexslider and discuss positives and negatives to their sue cases. This could include UX, maintenance, or any other aspect of a Silverstripe module being properly supported within the community.

**Keep in mind**

Almost all exiting Dynamic projects utilize Flexslider in some form or fashion, so a clearly defined transition plan will be required for an projects requiring an "upgrade" to anything other than this Flexslider module if the conclusion is to move away from this package.

**Comment by muskie9-opensource** [comment]
An alternative to Flexslider that we've been using quite a bit more is [Slick](http://kenwheeler.github.io/slick/). It seems to be lightweight and simple to implement.

That being said, a point of discussion we've had internally is whether a 3rd party plugin is necessary based on the abilities of Bootstrap. This may not be the css solution for all, so having a 3rd party option is nice to have to span those instances.

I do think that regardless of the solution, the model is usable (thinking of the 5, or soon to be 5 release) which allows for different types of slide records in a more consistent matter (text, image, video, etc) should be retained. The structure alone could speed up development on a project in a predictable manor. The frontend implementation _could_ be left up to the developer if no plugin is decided upon. Proper documentation would be required for this though.

Thinking to some other community modules, this could be an instance of "here's the default", but here's how you block some/all and do your own thing. This isn't inherently difficult in Silverstripe as long as the module takes it into consideration.

**Reply by NightJar** [reply]
Having a "default" library makes a lot of sense, and makes the module much more flexible (although it may need a rename :P)
The default slider/carousel should have great accessibility consideration too, IMO. There are many options that claim such available today, but few actually get implemented correctly to actually conform to standard. This is particaularly critical if introducing multimedia slides, most especially if they are intended to autoplay.

**Comment by lerni** [comment]
I’ve used flexslider in the past, but not this module. Almost always I use [swiper.js](https://github.com/nolimits4web/swiper) these days - it's well maintained, MIT licensed and offers integration with different JS lib's but most important, doesn't depend on jQuery as the other suggested one! Declarative parameters per data-attributes need some extra work, but I guess everything and more of what you need is possible.
